From stable+bounces-81209-greg=kroah.com@vger.kernel.org Mon Oct  7 05:59:33 2024
From: Jonathan Gray <jsg@jsg.id.au>
Date: Mon,  7 Oct 2024 14:59:22 +1100
Subject: Revert "drm/amd/display: Skip Recompute DSC Params if no Stream on Link"
To: gregkh@linuxfoundation.org
Cc: stable@vger.kernel.org
Message-ID: <20241007035922.55322-1-jsg@jsg.id.au>

From: Jonathan Gray <jsg@jsg.id.au>

This reverts commit d45c64d933586d409d3f1e0ecaca4da494b1d9c6.

duplicated a change made in 6.11-rc3
50e376f1fe3bf571d0645ddf48ad37eb58323919

Cc: stable@vger.kernel.org # 6.11
Signed-off-by: Jonathan Gray <jsg@jsg.id.au>
Signed-off-by: Greg Kroah-Hartman <gregkh@linuxfoundation.org>
---
 drivers/gpu/drm/amd/display/amdgpu_dm/amdgpu_dm_mst_types.c |    3 ---
 1 file changed, 3 deletions(-)

--- a/drivers/gpu/drm/amd/display/amdgpu_dm/amdgpu_dm_mst_types.c
+++ b/drivers/gpu/drm/amd/display/amdgpu_dm/amdgpu_dm_mst_types.c
@@ -1353,9 +1353,6 @@ static bool is_dsc_need_re_compute(
 	DRM_DEBUG_DRIVER("%s: MST_DSC check on %d streams in current dc_state\n",
 			 __func__, dc->current_state->stream_count);
 
-	if (new_stream_on_link_num == 0)
-		return false;
-
 	/* check current_state if there stream on link but it is not in
 	 * new request state
 	 */
